Managers/ConsoleMan.cpp

```diff
@@ -1,541 +1,423 @@
-
-void ConsoleMan::Update()
-{
-	bool inputConsumed = false;
-
-	if (g_UInputMan.FlagCtrlState() && g_UInputMan.KeyPressed(KEY_TILDE))
-	{
-		inputConsumed = true;
-		if (IsForceVisible())
-			ForceVisibility(false);
-		else
-			ForceVisibility(true);
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ConsoleMan::Destroy() {
+		SaveAllText("LogConsole.txt");
+
+		delete m_GUIController;
+		delete m_GUIInput;
+		delete m_GUIScreen;
+
+		Clear();
 	}
 
-    ////////////////////////////////////////////////////////////////////////
-    // Animate the menu into and out of view if enabled or disabled
-
-    if (m_EnabledState == ENABLING)
-    {
-        m_pParentBox->SetEnabled(true);
-        m_pParentBox->SetVisible(true);
-
-        float toGo = -floorf((float)m_pParentBox->GetYPos() * MENUSPEED);
-        m_pParentBox->SetPositionAbs(0, m_pParentBox->GetYPos() + toGo);
-
-        if (m_pParentBox->GetYPos() >= 0)
-            m_EnabledState = ENABLED;
-    }
-    // Animate the menu out of view
-    else if (m_EnabledState == DISABLING)
-    {
-        float toGo = -ceilf(((float)m_pParentBox->GetHeight() + (float)m_pParentBox->GetYPos()) * MENUSPEED);
-        m_pParentBox->SetPositionAbs(0, m_pParentBox->GetYPos() + toGo);
-
-        if (m_pParentBox->GetYPos() <= -m_pParentBox->GetHeight())
-        {
-            m_pParentBox->SetEnabled(false);
-            m_pParentBox->SetVisible(false);
-            m_EnabledState = DISABLED;
-        }
-    }
-    else if (m_EnabledState == ENABLED)
-    {
-        // If supposed to be enabled but appears to be disabled, enable it gracefully
-        if (m_pParentBox->GetYPos() < 0)
-            m_EnabledState = ENABLING;
-
-        m_pParentBox->SetEnabled(true);
-        m_pParentBox->SetVisible(true);
-        m_pInputTextBox->SetEnabled(true);
-        m_pInputTextBox->SetFocus();
-    }
-    else if (m_EnabledState == DISABLED)
-    {
-		if (!m_ForceVisible)
-		{
-			// If supposed to be disabled but appears to be out anyway, disable it gracefully
-			if (m_pParentBox->GetYPos() > -m_pParentBox->GetHeight())
-			{
-				m_pParentBox->SetEnabled(true);
-				m_pParentBox->SetVisible(true);
-				m_EnabledState = DISABLING;
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ConsoleMan::SetEnabled(bool enable) {
+		if (enable && m_ConsoleState != ENABLED && m_ConsoleState != ENABLING) {
+			m_ConsoleState = ENABLING;
+			g_GUISound.EnterMenuSound()->Play();
+		} else if (!enable && m_ConsoleState != DISABLED && m_ConsoleState != DISABLING) {
+			m_ConsoleState = DISABLING;
+			g_GUISound.ExitMenuSound()->Play();
+		}
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+	
+	void ConsoleMan::SetReadOnly() {
+		if (m_ReadOnly) {
```

Comment:
Given it's a simple condition and stuff, I'd reorganize this slightly to skip the early return.
I.e. 
```
if (!m_ReadOnly) {
    ...
}
```

Also, in the comment below I think `note` should be `mode`?